### PR TITLE
Use Ionic Storage

### PR DESCRIPTION
**Pull request by ihadeed**
This PR adds IonicStorage instead of the `SQLStorage` class. IonicStorage supports Indexedb, WebSQL, and SQLite. 

This adds `@ionic/storage` as a peer dependency. 

If you want me to send this PR to a different branch let me know.

### CODE REVIEW

**Comment by Nodonisko** [comment]
Is possible to add this as optional? Because if we merge this, we can't use this with Angular without Ionic.

**Comment by Nodonisko** [comment]
But it's not priority, so let's merge it.

File: src/cache.service.ts [hunk 1]
```diff
@@ -176,13 +171,23 @@ export class CacheService {
    */
   getItem(key: string): Promise<any> {
 
+    let nothingFound: boolean;
```
Comment (Nodonisko) [inline src/cache.service.ts]: Console log should not be on production

**Comment by Nodonisko** [comment]
Closes #13

**Comment by ihadeed** [comment]
Fixed!

It is possible to keep both implementations. It's still possible to use `@ionic/storage` in Angular 2 apps, there are no Ionic dependencies in that package. 

To keep both implementations, the current `CacheService` can be abstracted and then we can add two new services/modules. Each will implement the get/set/remove methods differently.

**Comment by Nodonisko** [comment]
I see, thanks.

**Pull request merged**